airbyte-integrations/connectors/source-salesforce-singer/source_salesforce_singer/source.py

```diff
@@ -74,22 +74,20 @@ def check_config(self, logger: AirbyteLogger, config_path: str, config: json) ->
     def discover_cmd(self, logger, config_path) -> str:
         return f"tap-salesforce --config {config_path} --discover"
 
-    def discover(self, logger: AirbyteLogger, config_container) -> AirbyteCatalog:
-        catalog = super().discover(logger, config_container)
-        return CatalogHelper.coerce_catalog_as_full_refresh(catalog)
-
     def read_cmd(self, logger, config_path, catalog_path, state_path=None) -> str:
         config_option = f"--config {config_path}"
         properties_option = f"--properties {catalog_path}"
-        return f"tap-salesforce {config_option} {properties_option} "
+        state_option = f"--state {state_path}" if state_path else ""
+        return f"tap-salesforce {config_option} {properties_option} {state_option}"
 
     def transform_config(self, raw_config):
-        # the select_fields_by_default is opinionated about schema changes. we want to reserve the right for the Airbyte system to handle these changes, instead of the singer source.
-        rendered_config = dict(raw_config)
-
-        # todo (cgardens) - this is supposed to be handled in the ui and the api but neither of them are able to handle it right now. issue: https://github.com/airbytehq/airbyte/issues/892
-        if "is_sandbox" not in raw_config:
-            rendered_config["is_sandbox"] = False
-
-        rendered_config["select_fields_by_default"] = True
+        # the select_fields_by_default is opinionated about schema changes. we want to reserve the right for the
+        # Airbyte system to handle these changes, instead of the singer source.
+        # todo (cgardens) - this is supposed to be handled in the ui and the api but neither of them are able to
+        #  handle it right now. issue: https://github.com/airbytehq/airbyte/issues/892
+        rendered_config = {
+            "is_sandbox": False,
```

Comment:
why is this defaulting to false? this should only be false if it is not already set in the raw_config.

Reply:
There is unpacking of raw configs in next row. If it contains is_sandbox parameter, the sandbox value will be changed to the value from raw_config. In another case it will stay False. Therefore the logic stays unchangeable - sandbox is  false if it is not already set in the raw_config. It is only written in shorter way.